br/pkg/lightning/backend/local/duplicate.go

```diff
@@ -965,15 +1074,16 @@ func (local *DupeController) CollectLocalDuplicateRows(ctx context.Context, tbl
 	if err != nil {
 		return false, errors.Trace(err)
 	}
-	if err := duplicateManager.CollectDuplicateRowsFromDupDB(ctx, local.duplicateDB, local.keyAdapter); err != nil {
+	if err := duplicateManager.CollectDuplicateRowsFromDupDB(ctx, local.duplicateDB, local.keyAdapter, algorithm); err != nil {
 		return false, errors.Trace(err)
 	}
 	return duplicateManager.HasDuplicate(), nil
 }
 
 // CollectRemoteDuplicateRows collect duplicate keys from remote TiKV storage. This keys may be duplicate with
 // the data import by other lightning.
-func (local *DupeController) CollectRemoteDuplicateRows(ctx context.Context, tbl table.Table, tableName string, opts *encode.SessionOptions) (hasDupe bool, err error) {
+// TODO: revise the returned arguments to (hasDupe bool, dupInfo *DupInfo, err error) to distinguish the conflict error and the common error
```

Comment:
```suggestion
// TODO: revise the returned arguments to (dupInfo *DupInfo, err error) to distinguish the conflict error and the common error
```
hasDup can be infered from `dupInfo != nil`

Reply:
Maybe not necessarily? Only when conflict detection mode is set to `error`, `CollectDuplicateRowsFromTiKV` would return `ErrFoundDataConflictRecords` or `ErrFoundIndexConflictRecords`. For other modes, we still need `duplicateManager.HasDuplicate()` to determine whether we have duplicate. And this information is needed for lightning
https://github.com/pingcap/tidb/blob/7548df70b1a4e84bd2ad2fa6cb52dfa819f4e5d3/br/pkg/lightning/importer/table_import.go#L1055-L1059